### PR TITLE
v0.9.0.7

### PR DESCRIPTION
**Pull request by jjr840430**

### CODE REVIEW

**Comment by jjr840430** [comment]
Fixed issue #16

File: tools/nos_to_cc.c [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	fprintf(src_fp, "#include \"%s\"\n\n", hdr_fname);
+	fprintf(src_fp, "static const char *g_nos_cfg = \"");
+
+	p = nos_str;
+	while (*p != '\0') {
+		if (*p == '"' || *p == '\n') {
+			fputc('\\', src_fp);
+		}
+		fputc(*p, src_fp);
+
```
Comment (drystone) [inline tools/nos_to_cc.c]: `\n` handling looks wrong to me - maybe something like:
```
if (*p == '\n')
    fprintf(src_fp, "\\n");
else {
    if (*p == '"')
        fputc("\\", src_fp);
    fputc(*p, strc_fp);
}
```

Comment (ronaldtse) [reply tools/nos_to_cc.c]: Agree

**Comment by ronaldtse** [comment]
Obviously @drystone 's review is much more workable, so I've updated the status now.

Especially on this part:
> I think you could do a lot more in terms of creating the nos structs in nos_to_cc instead of parsing the embedded string at runtime. Try making nereon_ctx_init take a pointer to an array of nereon_nos_options and see where that leads. Also nereon_parse_nos(const char *nos, struct nereon_nos_option **opts) may also be useful

@jjr840430 could you help update this? Thanks!

File: tools/nos_to_cc.c (hunk 1 above)
Comment (jjr840430) [reply tools/nos_to_cc.c]: @ronaldtse @drystone Fixed!

**Comment by jjr840430** [comment]
@ronaldtse @drystone 

From my thoughts, The `nereon_nos_options` structure will make to complicate users to use `libnereon` library.
I'm going to create new API function to get configurations from NOS, NOC and environment variables in batches.
So, the users may use it in more simple rather than manipulating `nereon_nos_options` structure when they use `libnereon` library in the programs.

Let me know your thoughts. Thanks!

**Comment by jjr840430** [comment]
@drystone Understand your thoughts.

So, do you want me to eliminate `rvd.nos` from the project and define configurations by the static structure?

I think It's not the original requirement of the project.

@ronaldtse What do you think about it?
If you allowed, then I will follow @drystone use cases.

**Comment by ronaldtse** [comment]
@drystone @jjr840430 the use cases are good.

We just want to ensure that:
1. A NOS parser parses the NOS (configuration schema) file
2. The NOS parser writes out a source code file (C, Go, whatever) that implements this proposed data structure.
3. The target application is written to directly use the NOS context, which was built in step 2.

Thanks!

**Comment by jjr840430** [comment]
@ronaldtse @drystone How about improvement in the future version?
I will merge it for now. Thanks!

**Comment by ronaldtse** [comment]
Absolutely. @jjr840430 can you also create a new issue with the aforementioned requests? Thanks!

**Pull request merged**